`XXH_ALIGN_MEMBER()` needs improvement

When we run the following command, it fails to build `collisionsTest`.

```sh
V=1 make clean usan
```

```
...
clang -Werror -O3 -I ./ -I ../../ -I allcodecs/  -c -o main.o main.c
clang -Werror -O3 -I ./ -I ../../ -I allcodecs/  -c -o pool.o pool.c
clang -Werror -O3 -I ./ -I ../../ -I allcodecs/  -c -o threading.o threading.c
clang++ -Wall -Wextra -Wconversion -std=c++11 -O3 -I ./ -I ../../ -I allcodecs/  -c -o sort.o sort.cc
In file included from sort.cc:34:
../../xxhash.h:983:4: error: 'alignas' attribute cannot be applied to types
   XXH_ALIGN_MEMBER(64, XXH64_hash_t acc[8]);
   ^
../../xxhash.h:937:47: note: expanded from macro 'XXH_ALIGN_MEMBER'
#   define XXH_ALIGN_MEMBER(align, type) type XXH_ALIGN(align)
                                              ^
../../xxhash.h:925:29: note: expanded from macro 'XXH_ALIGN'
#  define XXH_ALIGN(n)      alignas(n)
                            ^
...
```

This error only happens when we use `xxhash.h` with `clang++`.  Because

- It doesn't define `__STDC_VERSION__` for C++ code.
- It defines `__GNUC__`.
- But it doesn't accept GCC style attribute.

I don't remember older (pre-C++11 era) `clang++` accepts GCC style attribute or not.  But it's safe to exclude C++11 from the non-standard declaration.

```cpp
	/* Old GCC versions only accept the attribute after the type in structures. */
	#if !(defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L))   /* C11+ */ \
+	    && ! (defined(__cplusplus) && (__cplusplus >= 201103L)) /* >= C++11 */ \
	    && defined(__GNUC__)
	#   define XXH_ALIGN_MEMBER(align, type) type XXH_ALIGN(align)
	#else
	#   define XXH_ALIGN_MEMBER(align, type) XXH_ALIGN(align) type
	#endif
```

--- a/Makefile
+++ b/Makefile
@@ -175,6 +175,8 @@ clean:  ## remove all build artifacts
 	$(Q)$(RM) xxh32sum$(EXT) xxh64sum$(EXT) xxh128sum$(EXT)
 	$(Q)$(RM) $(XXHSUM_SRC_DIR)/*.o $(XXHSUM_SRC_DIR)/*.obj
 	$(MAKE) -C tests clean
+	$(MAKE) -C tests/bench clean
+	$(MAKE) -C tests/collisions clean
 	@echo cleaning completed
 
 

--- a/xxhash.h
+++ b/xxhash.h
@@ -933,6 +933,7 @@ struct XXH64_state_s {
 
 /* Old GCC versions only accept the attribute after the type in structures. */
 #if !(defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L))   /* C11+ */ \
+    && ! (defined(__cplusplus) && (__cplusplus >= 201103L)) /* >= C++11 */ \
     && defined(__GNUC__)
 #   define XXH_ALIGN_MEMBER(align, type) type XXH_ALIGN(align)
 #else

